utils/initial_data_preparation.py

```diff
@@ -3,17 +3,69 @@
 and preprocessing steps, and saves the results as CSV files. It also checks for
 null values and duplicate rows in the DataFrame.
 
-In addition to this, the rows from the original df where the
-code is hosted in the github.com domain is extracted and saved.
-
+In addition to this, the rows from the original DataFrame where the code is
+hosted in the github.com domain are extracted and saved separately.
+
+The script supports command-line arguments that allow users to specify
+custom paths for the input TSV file and the output directory for the CSV files.
+This enhancement makes the script more flexible and suitable for integrating
+into automated workflows where input and output paths may vary.
+
+Features:
+- Reads a TSV file specified by the user.
+- Performs data cleaning, including checking for null values and duplicates.
```

Comment:
Does the code also remove|reject the duplicates and rows with null values? I believe it should do so since it makes little sense to process these.